builtin/providers/digitalocean/resource_digitalocean_floating_ip.go

```diff
@@ -83,8 +83,10 @@ func resourceDigitalOceanFloatingIpRead(d *schema.ResourceData, meta interface{}
 	}
 
 	if _, ok := d.GetOk("droplet_id"); ok {
-		log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
-		d.Set("region", floatingIp.Droplet.Region.Slug)
+		if floatingIp.Droplet != nil {
+			log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
+			d.Set("region", floatingIp.Droplet.Region.Slug)
+		}
```

Comment:
@franklinhu the current tests for this are for both region based floatingips and floatingips with droplets. A floatingIP **must** either be created for a region OR a droplet. So if we fall back to the region then that seems to be the correct course of action

I haven't been able to trigger the panic